mlflow/store/model_registry/file_store.py

```diff
@@ -0,0 +1,767 @@
+
+    def __init__(self, root_directory=None):
+        """
+        Create a new FileStore with the given root directory.
+        """
+
+        super().__init__()
+        self.root_directory = local_file_uri_to_path(root_directory or _default_root_dir())
+        # Create root directory if needed
+        if not exists(self.root_directory):
+            mkdir(self.root_directory)
+
+    def _check_root_dir(self):
+        """
+        Run checks before running directory operations.
+        """
+        if not exists(self.root_directory):
+            raise Exception("'%s' does not exist." % self.root_directory)
+        if not is_directory(self.root_directory):
+            raise Exception("'%s' is not a directory." % self.root_directory)
+
+    def _validate_registered_model_does_not_exist(self, name):
+        model = self.get_registered_model(name)
+        if model is not None:
+            raise MlflowException(
+                f"Registered Model (name={name}) already exists.",
+                RESOURCE_ALREADY_EXISTS,
+            )
+
+    def _update_registered_model_last_updated_time(self, name, updated_time):
+        registered_model = self._get_registered_model(name)
+        registered_model.last_updated_timestamp = updated_time
+        registered_model_dict = dict(registered_model)
+        del registered_model_dict["tags"]
+        write_yaml(
+            join(self.root_directory, name),
+            FileStore.META_DATA_FILE_NAME,
+            registered_model_dict,
+            overwrite=True,
+        )
+
+    def create_registered_model(self, name, tags=None, description=None):
+        """
+        Create a new registered model in backend store.
+
+        :param name: Name of the new model. This is expected to be unique in the backend store.
+        :param tags: A list of :py:class:`mlflow.entities.model_registry.RegisteredModelTag`
+                     instances associated with this registered model.
+        :param description: Description of the model.
+        :return: A single object of :py:class:`mlflow.entities.model_registry.RegisteredModel`
+                 created in the backend.
+        """
+
+        self._check_root_dir()
+        _validate_model_name(name)
+        self._validate_registered_model_does_not_exist(name)
+        for tag in tags or []:
+            _validate_registered_model_tag(tag.key, tag.value)
+        meta_dir = mkdir(self.root_directory, name)
+        creation_time = now()
+        latest_versions = []
+        registered_model = RegisteredModel(
+            name=name,
+            creation_timestamp=creation_time,
+            last_updated_timestamp=creation_time,
+            description=description,
+            latest_versions=latest_versions,
+            tags=tags,
+        )
+        registered_model_dict = dict(registered_model)
+        # tags are stored under TAGS_FOLDER_NAME so remove them in meta file.
+        del registered_model_dict["tags"]
```

Comment:
Nit, instead of setting the tags on Line 140 and then deleting them here, can we pass `tags=None` on line 140? Can we apply similar changes for `create_model_version`?

Reply:
Unresolving until changes are pushed :)

Reply:
@serena-ruan I see. I don't think we can apply my suggestion because the returned `RegisteredModel` needs to contain tags.